main/routines.c

```diff
@@ -545,10 +555,24 @@ extern const char *baseFilename (const char *const filePath)
 	 */
 	for (i = 0  ;  i < strlen (PathDelimiters)  ;  ++i)
 	{
+#ifdef HAVE_MBLEN
+		const char *p;
+		int ml;
+
+		for (p = filePath  ;  *p != '\0'  ;  ++p)
+		{
+			ml = mblen(p, MB_LEN_MAX);
+			if (ml > 1)
+				p += ml - 1;
+			else if (*p == PathDelimiters [i] && p > tail)
+				tail = p;
+		}
+#else
```

Comment:
Hum, so it's a special case for one single encoding that happens to be "somewhat" ASCII compatible?  But then, what about e.g. UTF-16?  UTF-32?  IMO this would require real converting, knowing the input, and then working on a known result.